### PR TITLE
added report functionality, reservation is wip but xml is done

### PR DESCRIPTION
**Pull request by nbaggett**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -25,20 +25,26 @@ android {
...
 
     // Firebase
     implementation 'com.google.firebase:firebase-core:16.0.9'
     implementation 'com.google.firebase:firebase-database:17.0.0'
     def lifecycle_version = "2.0.0"
-    // ViewModel and LiveData
-    implementation "androidx.lifecycle:lifecycle-extensions:$lifecycle_version"
+    implementation 'androidx.lifecycle:lifecycle-extensions:2.0.0'
 
+    implementation 'com.google.code.gson:gson:2.6.2'
```
Comment (LucasYoung) [inline app/build.gradle:38]: This is unused

File: app/src/main/java/polyrooms/polyrooms/MainActivity.kt [hunk 2]
```diff
@@ -17,19 +18,56 @@ import androidx.core.content.ContextCompat
...
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         requestWindowFeature(Window.FEATURE_NO_TITLE)
         window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                 WindowManager.LayoutParams.FLAG_FULLSCREEN)
         setContentView(R.layout.activity_main)
+
+        val sharedPreferences = getSharedPreferences("production", Context.MODE_PRIVATE)
+
+        //STORE A CUSTOM OBJECT TO SHARED PREFERENCES
```
Comment (LucasYoung) [inline app/src/main/java/polyrooms/polyrooms/MainActivity.kt:49]: Unused code

File: app/src/main/java/polyrooms/polyrooms/ReportActivity.kt [hunk 3]
```diff
@@ -2,16 +2,74 @@ package polyrooms.polyrooms
...
+        } else if (!checkRoomNum(roomNum)) {
+            Toast.makeText(this, "You must enter a room number.", Toast.LENGTH_LONG).show()
+        } else if (!checkDescription(description)) {
+            Toast.makeText(this, "You must enter a description.", Toast.LENGTH_LONG).show()
+        } else { // everything gucci time to submit
+            val report: Report = Report(description)
+
+            reportRoom(buildingNum, roomNum, report)
+            Toast.makeText(this, "Thank you, your report has been sent.", Toast.LENGTH_LONG).show()
+            /*
```
Comment (LucasYoung) [inline app/src/main/java/polyrooms/polyrooms/ReportActivity.kt:36]: Unused code

**Review by LucasYoung — CHANGES_REQUESTED** [review]
The logic for reporting the room is correct. This just needs to be cleaned up before being merged.

File: app/build.gradle (hunk 1 above)
Comment (nbaggett) [reply app/build.gradle:38]: This is used for shared preferences

File: app/src/main/java/polyrooms/polyrooms/MainActivity.kt (hunk 2 above)
Comment (nbaggett) [reply app/src/main/java/polyrooms/polyrooms/MainActivity.kt:49]: See above

File: app/src/main/java/polyrooms/polyrooms/ReportActivity.kt (hunk 3 above)
Comment (nbaggett) [reply app/src/main/java/polyrooms/polyrooms/ReportActivity.kt:36]: This is a message to the user

File: app/src/main/java/polyrooms/polyrooms/MainActivity.kt (hunk 2 above)
Comment (LucasYoung) [reply app/src/main/java/polyrooms/polyrooms/MainActivity.kt:49]: This comment refers to the commented out code

File: app/src/main/java/polyrooms/polyrooms/ReportActivity.kt (hunk 3 above)
Comment (LucasYoung) [reply app/src/main/java/polyrooms/polyrooms/ReportActivity.kt:36]: This is the commented out code

**Pull request closed without merging**